### ISSUE TITLE
No entry for clearing playback history

### ISSUE THREAD

**Issue by Yelfive** [lhc70000/iina#890]
**System and IINA version:**

- macOS 10.12.5
- IINA 0.0.11

**Expected behavior:**

I sometimes need to clear the playback history, but there's no entry for that, hopping that's in your schedule, thanks

**Comment by oscar-c** [comment]
- Select first entry;
- scroll to last one;
- select last one while holding down <kbd>Shift</kbd>;
- Right click on any highlighted item;
- Select "Delete...".

Although this is a workaround, the support using <kbd>⌘</kbd>+<kbd>A</kbd> to select all, and <kbd>Delete</kbd> to delete would be also helpful and potentially a solution for this problem.

**Comment by Yelfive** [comment]
@oscar-c actually, I found the `delete` in the context menu of right click, but that's in `disabled` state, and cannot be clicked, and the keyboard `delete` does not response

P.S. how do you render a button in github comment

**Comment by lhc70000** [comment]
@Yelfive You must select something before deleting

**Comment by Yelfive** [comment]
@lhc70000 Yes, I have selected items

![image](https://user-images.githubusercontent.com/7898026/28418720-50ce4a60-6d8f-11e7-9cea-057079f5b96f.png)

**Comment by oscar-c** [comment]
Try upgrade to the latest version 0.0.12.
It should look like this ![12](http://d.pr/i/H2QUWD+)

To make the keyboard glyph, you can use the `<kbd></kbd>` tag.

**Comment by Yelfive** [comment]
@oscar-c  Thx , I thought mime is the newest.
However, after upgrading to 0.0.12, <kbd>delete</kbd>'s also disabled. I might opened IINA in the wrong way [FACEPALM]
![image](https://user-images.githubusercontent.com/7898026/28446587-46b8c342-6dfe-11e7-8fa4-48146bc3fc5b.png)

**Comment by Yelfive** [comment]
I have waited for the 0.0.14 finally, but I still cannot delete the playback history.

Is there any way else I can take to clear the history?

**Comment by lhc70000** [comment]
@Yelfive Select all (use shift click, or cmd+A when there's no playing video) and right click - delete

**Comment by Yelfive** [comment]
@lhc70000 yes I have tried that, same as 0.0.12, the <kbd>delete</kbd> is disabled as well

![image](https://user-images.githubusercontent.com/7898026/32142687-4e9653da-bc6a-11e7-8f31-44be5413c729.png)

**Comment by lhc70000** [comment]
Never seen this before, please delete `~/Library/Application Support/com.colliderli.iina/history.plist` directly if you really need.

Is the "delete" menu item always disabled on your machine?

**Comment by Yelfive** [comment]
Thanks, deleting the `history.plist` works out for me, and yes, `delete` is always disabled and my OSX version is 10.11.6

**Comment by iammeat** [comment]
The issue seems to occur where delete is disabled if the original file has been deleted from the hard drive.  Can you add a fix for this?

**Comment by iammeat** [comment]
I am also using 10.11.6 and the delete option is also always disabled.

**Comment by marco-schmidt** [comment]
- This feature should be more obvious.
- For me, deleting a small number of items like described in this ticket takes a minute. If I select all entries, the program does not react anymore and I have to abort it at some point. Deletion of all entries does not seem to be efficient for a larger history.
- The workaround of deleting the plist file works for me, but is hardly intuitive.
- The delete feature in the menu for the most recently used documents (cannot find the English menu labels, sorry) suggests that the history **is** being deleted, at least for people who have no idea the 'history list' feature (cmd-shift-H) exists. I think that the deletion feature should ask if the entire playback history should be deleted, otherwise a person might think they deleted it when in fact they didn't. This would give more control to users.

Thanks for creating and maintaining IINA.
